lib/Dialect/TTNN/IR/TTNNOpsAttrs.cpp

```diff
@@ -453,14 +495,24 @@ TTNNLayoutAttr TTNNLayoutAttr::get(
     Type elementType, BufferType bufferType, GridAttr grid,
     TensorMemoryLayoutAttr memLayoutAttr,
     ArrayRef<std::pair<std::int64_t, std::int64_t>> collapseIntervals) {
+
   // Construct a new affine map which will be used to map from logical
-  // space to physical space
+  // space to physical space.
   AffineMap linear = collapsedLinearAffineMap(
       context, tensorShape, grid.getShape(), collapseIntervals);
-  // Calculate shard shape by evaluating the linear map with last element
-  // of the tensor shape and dividing it by the grid shape
-  mlir::SmallVector<int64_t, 4> shardShape =
-      calculateLogicalShardShape(tensorShape, linear, grid);
+
+  // Calculate shard shape by evaluating the linear map.
+  //
+  mlir::SmallVector<int64_t> shardShape;
+  if (bufferType == BufferType::L1 &&
```

Comment:
Why does this only applies to L1? Wouldn't it be the same for DRAM?

Reply:
We decided to not include DRAM since we do not have a correct way of modelling grid shape for DRAM. Currently, grid is only meaningfully related to L1 memory.

Reply:
DRAM banks live in their own cores, separate from the tensix grid. My view is that the GridAttr is totally irrelevant when buffer type is DRAM. Maybe a proper way of modeling this would be to have GridAttr set to some none value in this case?

But it's not the scope of this PR

Reply:
I see. Lets continue discussion in #1628.